### ISSUE TITLE
Support new aws_lb resource?

### ISSUE THREAD

**Issue by gmccue** [terraform-aws-modules/terraform-aws-alb#12]
It looks like Terraform is moving to the `aws_lb` resource, and several new features have been added in rel `1.1.0` https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#110-october-16-2017

I would be happy to submit a PR to integrate these changes.

**Comment by brandoconnor** [comment]
Good catch @gmccue ! I'll be keeping a closer watch on the upstream repo here on in.

That set of name changes on the resources/data sources also likely means the network load balancer could be handled in this project with a simple bit flip ([tf docs are a little sparse on it thus far](https://www.terraform.io/docs/providers/aws/r/lb.html#load_balancer_type)). Let's leave that for another issue/PR but keep that in mind when building.

Feel free to take on the resource name change in a PR; I'm happy to review. We have until the next major release of the AWS provider before the current resource names are deprecated.

**Comment by gmccue** [comment]
I have a PR ready to go for this, but it's blocked by https://github.com/terraform-providers/terraform-provider-aws/pull/2256 at the moment. 
I'll submit once the bug fix is implemented in `terraform-provider-aws`.

**Comment by brandoconnor** [comment]
The v.3.0.0 release uses an aws_lb resource. Given the trickiness that you've also seen around NLB, this module isn't able to do both ALB and NLB just yet. I wonder if the provider will end up separating that resource out after all. In either case, I think this can be closed. I'll keep an eye on opportunity to roll in NLB.
